polymer/src/sample-category-data.html

```diff
@@ -0,0 +1,207 @@
+        var catNames =  this.resource.map(function(cat) {
+          var obj = {
+            name:  cat["name"],
+            id: cat["id"]
+          };
+          return obj;
+        });
+        this.categories = catNames;
+      }
+
+      _getCategoryObject(categoryName, categories) {
+        if (categories === undefined) {
+          return null;
+        }
+        for (let i = 0, c; c = categories[i]; ++i) {
+          if (c.id === categoryName) {
+            this.category = c;
+            return c;
+          } else if (categoryName === "index.html") {
+            return null
+          }
+        }
+      }
+
+      _computeCategory(categoryName) {
+
+        // Fetch the items of the category. Note that the fetch is asynchronous,
+        // which means `category.items` may not be set initially (but that path
+        // will be notified when the fetch completes).
+        let categoryObj = this._getCategoryObject(categoryName, this.categories);
+        this._fetchItems(categoryObj, 1);
+        return categoryObj;
+      }
+
+      _computeItem(items, itemName) {
+        if (!items || !itemName) {
+          return;
+        }
+        for (let i = 0, item; item = items[i]; ++i) {
+          if (item.url.endsWith(itemName.concat('/'))) {
+            return item;
+          }
+        }
+      }
+
+      _parseCategoriesFromJson(e) {
+        return JSON.parse(e.target.responseText)[0]["categories"]
+      }
+
+      _fetchCategories(attempts) {
+        this._setFailure(false);
+        if (this.categories) {
+          return;
+        }
+        this._getResource({
+          url: SampleCategoryData.sitemapUrl,
+          onLoad(e) {
+            this.resource = this._parseCategoriesFromJson(e)
+            var catNames =  this.resource.map(function(cat) {
+              var obj = {
+                name:  cat["name"],
+                id: cat["id"]
+              };
+              return obj;
+            });
+            this.set('categories', catNames);
+          },
+          onError(e) {
+            this._setFailure(true);
+          }
+        }, attempts);
+      }
+
+      _fetchItems(category, attempts) {
+        this._setFailure(false);
+        // Only fetch the items of a category if it has not been previously set.
+        if (!category || category.items) {
+          return;
+        }
+
+        var cats = this.resource
+        cats = cats.filter(function(cat) {
+                return cat.name == category.name
+        });
+
+        var items =  cats[0]["examples"].map(function(item) {
+        var obj = {
+          name:  item["name"],
+          title: item["title"],
+          description: item["description"],
+          url: item["url"]
+        };
+        return obj;
+        });
+        this.set('category.items', items);
+        category.items = items
+      }
+
+      _getResource(rq, attempts) {
+        let xhr = new XMLHttpRequest();
```

Comment:
Is there a particular reason why you're using an XMLHttpRequest here? You're using the fetch API above...

Reply:
removed the entire method

Reply:
It's still here..